File: examples/telephone/third/kid.rb
require 'dramatis/actor'

require 'mangler'

class Kid

  include Dramatis::Actor
  
  def initialize name, next_kid = nil
    @name = name
    @next = next_kid
  end

  def to_s
    @name
  end

  def whisper what
    @heard = mangle what
    @next and @next.whisper @heard
  end

  def ask
    @heard
  end

  def mangle what
    Mangler.mangle what
  end

end

File: examples/telephone/third/kid.py
import dramatis

from  mangler import Mangler

class Kid( dramatis.Actor ):
  
  def __init__(self, name, next_kid = None):
      super(Kid,self).__init__()
      self._name = name
      self._next = next_kid

  def __str__(self):
      return self._name
  
  def mangle(self, what):
      return Mangler.mangle( what )

  def whisper(self, what):
      self._heard = self.mangle( what )
      if self._next:
          self._next.whisper( self._heard )

  def ask(self): return self._heard

